README.md

```diff
@@ -15,251 +18,329 @@ The role can be used to configure:
+For `initscripts`, the legacy network service is required as used in Fedora or RHEL.
 
-- For initscripts, profiles correspond to ifcfg files in `/etc/sysconfig/network-scripts/ifcfg-*`.
+- For `initscripts`, profiles correspond to ifcfg files in the `/etc/sysconfig/network-scripts/ifcfg-*` directory.
 
-- For NetworkManager, profiles correspond to connection profiles as handled by NetworkManager.  Fedora and RHEL use the `rh-plugin` for NetworkManager which also writes configuration files to `/etc/sysconfig/network-scripts/ifcfg-*` for compatibility.
+- For `NetworkManager`, profiles correspond to connection profiles as handled by
+  NetworkManager.  Fedora and RHEL use the `ifcfg-rh-plugin` for NetworkManager,
+  which also writes or reads configuration files to `/etc/sysconfig/network-scripts/ifcfg-*`
+  for compatibility.
 
-Note that the role primarily operates on networking profiles (connections) and
-not on devices but it defaults to use the profile name as the interface name.
-But it is also possible to create generic profiles, by creating for example a
+Note that the `network` role primarily operates on networking profiles (connections) and
+not on devices, but it uses the profile name by default as the interface name.
+It is also possible to create generic profiles, by creating for example a
 profile with a certain IP configuration without activating the profile. To
-apply the configuration to the actual networking interface, a command like
-`nmcli` needs to be used on the target system.
+apply the configuration to the actual networking interface, use the `nmcli`
+commands on the target system. For each host, a list of networking profiles can
+be configured via the `network_connections` variable.
 
-### Warning
+**Warning**: The `network` role updates or creates all connection profiles on
+the target system as specified in the `network_connections` variable. Therefore,
+the `network` role removes options from the specified profiles if the options are
+only present on the system but not in the `network_connections` variable.
 
-The role updates or creates all connection profiles on the target system as
-specified in the `network_connections` variable. Therefore, the role will
-remove settings from the specified profiles if the settings are only present on
-the system but not in the `network_connections` variable. The following
-exceptions apply:
+Variables
+---------
+The `network` role is configured via variables starting  with  `network_` as the name prefix.
+List of required variables:
 
-* For profiles that only contain a `state` setting, the role will only activate
-  or deactivate the connection without changing its configuration.
+* `network_provider` - The `network_provider` variable allows to set a specific
+  provider (`nm` or `initscripts`) . Setting it to `network_provider_os_default`,
+  the provider is set depending on the operating system. This is usually `nm`
```

Comment:
This is confusing. It does not say that `network_provider_os_default` is a fact set by the role. But this problem existed in the original version as well